catalog/schematron/catalog-schematron-plugin/src/test/java/ddf/services/schematron/SchematronValidationServiceTest.java

```diff
@@ -166,6 +167,26 @@ public void testWithSpacesInPathToSchematronFile()
                 "dog_3leg_3paw.xml"));
     }
 
+    @Test
+    public void testWarningsAndErrorsAreSanitized()
+            throws ValidationException, IOException, SchematronInitializationException {
+        SchematronValidationService service = getService(false,
+                "doggy-namespace",
+                true,
+                "dog_legs.sch",
+                "dog_paws.sch");
+        MetacardImpl metacard = getMetacard("dog_4leg_4paw_namespace.xml");
+        service.validate(metacard);
+        assertThat(service.getSchematronReport(), is(notNullValue()));
+        Pattern pattern = Pattern.compile("[\t \r\n]+");
+        for (String warning : service.getSchematronReport().getWarnings()) {
+            assertThat(pattern.matcher(warning).matches(), is(false));
```

Comment:
Have you checked that this matcher is actually testing what you're trying to test? It's not. You can use a scratch project to test things like this.

What you're looking for is this:
    for (String s : strings) {
        s.matches(".*[\t \r\n]+.*");
    }